src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
```

Comment:
I suggest using a `reduce` function here, for ex:

```  
const findMotherBornDate = (motherName) =>
    people.find((mother) => mother.name === motherName).born;

const sum = children.reduce(
    (counter, ch) => (counter += ch.born - findMotherBornDate(ch.mother)), 0
  );
```
And as I mention above you can return a variable to make it more readable